Treat .csx/.vbx as regular source files from command-line compilers

cston opened the pull request:
Treat .csx/.vbx as regular source files from command-line compilers

As a result, a significant number of unit tests are now skipped. (Search for SCRIPTING define and https://github.com/dotnet/roslyn/issues/2888.)

cston commented:
@tmat, @ManishJayaswal, @Pilchie, @jaredpar

tmat commented:
:+1:

cston commented:
Bug: #1979

Customer scenario: C# and VB script syntax should not be supported: .csx and .vbx files should be treated as regular C# and VB source rather than as script source; and from the Compiler API, it should not be possible to parse a document as a script.

Fix: ".csx" and ".vbx" extensions are no longer handled differently by the CommandLineParser. And ParseOptions.WithKind(SourceCodeKind) is now internal rather than public. Note: There are a significant number of Editor unit tests that test Script and Interactive documents. Since ParseOptions.WithKind() is internal, those tests are now skipped.

Testing: Test pass

Pilchie commented:
Looks good, thanks!

ManishJayaswal commented:
We are changing public APIs. Can we make sure this is approved by compat council and send breaking change notification? GTM otherwise.

MattGertz commented:
Approved 5/21 by ML Shiproom.  @kuhlenh can you consider the PSA on this -- we are going to keep it on in the go-ahead branch, it's just off in this one only.

cston commented:
Changed CSharpParseOptions and VisualBasicParseOptions constructors to throw NotSupportedException if kind != SourceCodeKind.Regular. And changed DocumentState.UpdateSourceCodeKind to throw NotSupportedException if SourceCodeKind is changed.

tmat commented:
:+1:

cston commented:
Fixes #1979

BobStrogg commented:
Maybe a dumb question, but why was this changed?  I'm using Roslyn to compile simple inline expressions, and SourceCodeKind.Interactive seemed to work pretty well for that.  Is there an alternative way I can do this, or do I need to wrap it all in a C# class definition?  Thx :)

Daviziu commented:
I second @BobStrogg's question
I used to use SourceCodeKind.Interactive in a similar manner as well.

What I'd like to know is: Are scripts with top-level expressions and statements still supported? If so, how can these sources be parsed if SourceCodeKind.Interactive cannot be used? 
Or, could it be that scripts like this are not supported and never will be supported again? In this case I'd have to make a lot of changes to my code.

Daviziu commented on src/Compilers/CSharp/Portable/CSharpParseOptions.cs:
```diff
@@ -50,6 +50,13 @@ public override IEnumerable<string> PreprocessorSymbolNames
+#if !SCRIPTING
+            if (kind != SourceCodeKind.Regular)
+            {
+                throw new NotSupportedException(kind.ToString());
```
Why was this changed and is there any workaround that still allows parsing of interactive script sources?

KevinH-MS commented:
@BobStrogg This change was only made in the ```stabilization``` branch (bits that went into VS 2015 RTM).  Scripting/Interactive is still very much supported (and currently under active development) in the ```master``` branch.

tmat commented:
Yes.
